fmp_server.py

```diff
@@ -305,18 +305,51 @@ def recvReqLayerUp(self, packet_id, data, seq):
          - Relocate > Select Server
         """
         if self.session.layer == 2:
-            city = self.session.get_city()
-            if city.leader == self.session:
-                if len(city.players) > 1:
-                    # TODO: Transfer the leadership to another player
-                    pass
-                city.leader = None
+            new_leader = self.session.try_transfer_city_leadership()
```

Comment:
I would have preferred a generic "layer" transfer of ownership something like `layer_migrate_host` but let's keep it like this for now.